src/library/scala/collection/Iterable.scala

```diff
@@ -713,6 +713,22 @@ object IterableOps {
 @SerialVersionUID(3L)
 object Iterable extends IterableFactory.Delegate[Iterable](immutable.Iterable) {
   implicit def toLazyZipOps[A, CC[X] <: Iterable[X]](that: CC[A]): LazyZipOps[A, CC[A]] = new LazyZipOps(that)
+
+  /**
+    * @return an Iterable by using a function `f` producing elements of
+    *         type `A` and updating an internal state `S`.
+    * @param init State initial value
+    * @param f    Computes the next element (or returns `None` to signal
+    *             the end of the collection)
+    * @tparam A   Type of the elements
+    * @tparam S   Type of the internal state
+    */
+  def unfold[A, S](init: S)(f: S => Option[(A, S)]): Iterable[A] = {
+    val initialState = init // unfortunately, `Iterable` has an `init` method
```

Comment:
This renaming bugs me, but I'm honestly not sure what the best course of action is.

`init` is a good, short parameter name, and I'm not a huge fan of a longer name like `initialState`. Ideally, all of the `unfold` methods ought to have the same parameter names, so changing this one means changing `LazyList.unfold` and `Iterator.unfold` as well.

Perhaps lengthen it slightly to `initial`?

Let the bikeshedding commence!

Reply:
What if you `@inline def initialState= init`?